project/npda/views/home.py

```diff
@@ -51,20 +56,37 @@ def home(request):
     """
     if request.method == "POST":
         form = UploadFileForm(request.POST, request.FILES)
-        file = request.FILES["csv_upload"]
+        user_csv = request.FILES["csv_upload"]
         pz_code = request.session.get("pz_code")
 
-        # summary = csv_summarize(csv_file=file)
+        # You can't read the same file twice without resetting it
+        user_csv.seek(0)
+        reader = csv.reader(user_csv.read().decode("utf-8").splitlines())
+        user_headers = next(reader, None)
+
+        template_headers = HEADINGS_LIST
+
+        # Next localise headers so have list and say 'these are the headers affected'
+
+        if user_headers != template_headers:
+            list_diff = [header for header in user_headers if header not in template_headers]
+            error_string = f'CSV headers do not match the expected format/order. Please ensure you are using the template csv provided. Headers affected, or those present in your file but not the template, are: {list_diff}. The headers uploaded must be exactly the same as the template headers, including whitespace, capital letters and ordering.'
```

Comment:
I don't think the order matters? I agree about trimming whitespace on the header names